server/custom/reports/controller.js

```diff
@@ -17,8 +17,73 @@ exports.ReportsFindAll = async function (req, res) {
     req.query.companyid = true;
     req.user = {};
     req.user.companyID = 'COMPID';
-    const result = await controller.findAll(req);
-    res.status(200).json(result);
+    var perPage = config.get('pagination.itemsPerPage');
+    var page = (req.query.page) ? req.query.page : 1;
+    var result = controller.findAllComponent(req);
+    var count = await Reports.count(result.find).exec();
+    if (req.query.populate === 'layer') {
+        var aggregation = Reports.aggregate([
+            {
+                $lookup:
+        {
+            from: 'wst_Layers',
+            localField: 'selectedLayerID',
+            foreignField: '_id',
+            as: 'layer'
+        }
+
+            },
+            {
+                $unwind:
+     {
+         path: '$layer',
+     }
+            },
+            {
+                $match: result.find,
+            },
+            {
+                $project: {
+                    'result.fields': 1,
+                    selectedLayerID: 1,
+                    properties: 1,
+                    reportType: 1,
+                    query: 1,
+                    reportName: 1,
+                    owner: 1,
+                    author: 1,
+                    createdBy: 1,
+                    createdOn: 1,
+                    companyID: 1,
+                    nd_trash_deleted: 1,
+                    history: 1,
+                    __v: 1,
+                    isPublic: 1,
+                    layerName: '$layer.name'
+                }
```

Comment:
Are all these fields really needed ? Is `__v` used anywhere for instance ?

You should probably fetch only fields listed in `result.fields` (that comes from `findAllComponent`) + `layerName`